Dropdowns with multiple select

geek-yang opened the pull request:
Add dropdowns with multiple select feature. We use the external plugin [vue-multiselect](https://www.npmjs.com/package/vue-multiselect).

The "past performance" tab and "future change" tab are merged into one tab.

The single select dropdown component will be replaced by this plugin as well (not that it is not deleted since "Analyse" tab still uses it. It can be removed later).

This PR closes #43 and fixes #54.

geek-yang commented:
This PR is ready for review. Please focus on the functionality of multiple dropdown, which allows the user to play with the display of cpm/rcm/gcm maps and past performance/future change experiments.

The styling of the page is not a primary concern in this PR. Given that this PR involves lots of changes already, it is better to address it in another PR, or together with the re-design of the whole cpm page.

bvreede reviewed (approved):
Approving bc gh is being a *****

bvreede commented:
The name "performance" works, but I would argue for the name "Predictions" instead (or "performance & predictions", but that might be too long), as the main outcome of interest (? at least for me) from these models is their predictions for future change. Having the performance there is valuable, but it is not why the models were developed.

Peter9192 commented:
Is it possible to merge main into this branch to get rid of all the figure changes?

geek-yang commented:
> Is it possible to merge main into this branch to get rid of all the figure changes?

Good point! I thought I already did, but that was before we merged the PR that changed all the figure names. Thanks for suggesting!

Peter9192 commented on components/cpm/Performance.vue:
```diff
@@ -1,54 +1,119 @@
+          track-by="name"
+          :preselect-first="true"
+        >
+          <template slot="selection" slot-scope="{ values, isOpen }"><span v-if="values.length &amp;&amp; !isOpen" class="multiselect__single">{{ values.length }} modelling system(s) selected</span></template>
```
Modelling systems is not clear and also not really accurate. I think I'd prefer either:

- Projects (CMIP, CORDEX, CPM)  (this is accurate)
- Type (global, regional, convection-permitting)   (this holds the middle)
- Resolution (low, medium, high)   (this is easier to understand)

geek-yang replied:
Thanks for your suggestions. I prefer the accurate one, the Projects (CMIP, CORDEX, CPM). They are implemented now.

Peter9192 commented on components/Navigation.vue:
```diff
@@ -44,7 +44,7 @@ export default {
           url: '/cpm',
           children: [{ title: 'Explore', url: '/cpm/explore' },
-            { title: 'Past performance', url: '/cpm/performance' },
+            { title: 'Performance', url: '/cpm/performance' },
```
I think the page with past performance/future change should be called "explore". The page that's now called 
"explore" can be changed to "Lines of evidence" or so.

geek-yang replied:
The tabs are renamed accordingly 😃.

Peter9192 reviewed (approved):
Purely focusing on the functionality, I tested this and can confirm it works. The layout and also the code can be improved, but let's do that in follow-up PRs. 

Can we split the dropdowns and figure display into separate components?

Please verify that the live site also works fine after merging this PR

geek-yang commented:
> Purely focusing on the functionality, I tested this and can confirm it works. The layout and also the code can be improved, but let's do that in follow-up PRs.
> 

@Peter9192 Thanks for your feedback! Just finish addressing your comments. I will go ahead and merge it.

> Can we split the dropdowns and figure display into separate components?
> 

I thought about it before. But since the figure display is only an div with a background figure, it feels a bit over-engineering to wrap it up as a single component. I would suggest we leave it like this and get back to it when we want to polish the code for the whole site, then we can have a better vision about which part could be reused (and make it a new component).


> Please verify that the live site also works fine after merging this PR
> 

No problem. I will keep an eye on it.
